src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -104,12 +104,13 @@ source_locationt make_function_pointer_restriction_assertion_source_location(
   return source_location;
 }
 
-template <typename Handler>
-void for_each_function_call(goto_functiont &goto_function, Handler handler)
+template <typename Handler, typename GotoFunctionT>
+void for_each_function_call(GotoFunctionT &&goto_function, Handler handler)
```

Comment:
:question: Why does this become generic?

Reply:
This is so that the function can take both const and non-const goto function objects.

Reply:
I don't follow - you can pass a non-const version of something to a function that takes a const object? I'll try undoing this change locally and see if then it clicks!

Reply:
Oh presumably because the handler might require a non-const version.